testing/deprecated_test.py

```diff
@@ -90,3 +90,34 @@ class MockConfig:
         nodes.Node(name="test", config=ms, session=ms, nodeid="None")
     assert w[0].lineno == inspect.currentframe().f_lineno - 1
     assert w[0].filename == __file__
+
+
+@pytest.mark.filterwarnings("default")
+@pytest.mark.parametrize("config_mode", ["ini", "cmdline"])
+def test_noprintlogs_is_deprecated_ini(testdir, config_mode):
+    if config_mode == "ini":
+        args = ()
+        testdir.makeini(
+            """
+            [pytest]
+            log_print=False
+            """
+        )
+    else:
+        assert config_mode == "cmdline"
+        args = ("--no-print-logs",)
```

Comment:
I tend to try and avoid logic in tests, I'd probably write this as two separate tests instead of this

tests should be as simple as possible for the programmer to read

here's a blog post I often link on this subject (a little more severe of a case there, but still the same sentiment): https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

Reply:
Originally @vinaycalastry wrote two separate tests, and it was I who asked him to unify in a single one.

I agree we should avoid too much logic in tests in general, but I tend to parametrize tests when only the setup logic is involved (like here).

Reply:
Btw if you feel strongly here I'm perfectly fine to change it back to two tests, as I also appreciate the simplicity, although I'm not a big fan of the copy/paste involved then.

Reply:
I'd rather see the assertion as a helper function than a stringly typed if statement -- no need for duplication and it's very clear the execution path

Reply:
I can add a helper function and amend my commit.

Reply:
Changed to use a helper function. Please review

Reply:
I meant for a helper for the bottom half and don't use parametrize or an if statement

Reply:
(@asottile I suggest using the "request changes" feature so it is explicit that this PR is awaiting further changes 👍)

Reply:
Changed it.

Reply:
Please review